libraries/microsoft-agents-hosting-core/microsoft_agents/hosting/core/app/agent_application.py

```diff
@@ -204,11 +203,54 @@ def options(self) -> ApplicationOptions:
         """
         return self._options
 
+    def add_route(
+        self,
+        selector: RouteSelector,
+        handler: RouteHandler[StateT],
+        is_invoke: bool = False,
+        is_agentic: bool = False,
+        rank: RouteRank = RouteRank.DEFAULT,
+        auth_handlers: Optional[list[str]] = None,
+    ) -> None:
+        """Adds a new route to the application.
+
+        Routes are ordered by: is_agentic, is_invoke, rank (lower is higher priority), in that order.
+
+        :param selector: A function that takes a TurnContext and returns a boolean indicating whether the route should be selected.
+        :type selector: RouteSelector
+        :param handler: A function that takes a TurnContext and a TurnState and returns an Awaitable.
+        :type handler: RouteHandler[StateT]
+        :param is_invoke: Whether the route is for an invoke activity, defaults to False
+        :type is_invoke: bool, optional
+        :param is_agentic: Whether the route is for an agentic request, defaults to False. For agentic requests
+            the selector will include a new check for `context.activity.is_agentic_request()`.
+        :type is_agentic: bool, optional
+        :param rank: The rank of the route, defaults to RouteRank.DEFAULT
+        :type rank: RouteRank, optional
+        :param auth_handlers: A list of authentication handler IDs to use for this route, defaults to None
+        :type auth_handlers: Optional[list[str]], optional
+        :raises ApplicationError: If the selector or handler are not valid.
+        """
+        if not selector or not handler:
+            logger.error(
+                "AgentApplication.add_route(): selector and handler are required.",
+                stack_info=True,
+            )
+            raise ApplicationError("selector and handler are required.")
+
+        if is_agentic:
+            selector = _agentic_selector(selector)
+        route = _Route[StateT](
+            selector, handler, is_invoke, rank, auth_handlers, is_agentic
```

Comment:
If you're passing in both the selector and the "is_agentic", do you need to do the wrapper above? That is, inside the route method you could create the _agentic_selector as needed. Pushing that logic down further seems like it would simplify this method.